meetings/minutes/2025-03-03-TSC-Meeting.md

```diff
@@ -0,0 +1,131 @@
+
+Adding packages is done by editing the `ecosystem-packages.json` file.
+At the moment there's no limit on how many packages can be added at once.
+
+The process is not automated, with users having to create pull requests that must be reviewed by maintainers.
+
+Abandoned packages are skipped by default - should this still be the case?
+
+Removing packages is currently done by deleting the entry and submitting a PR, with no other steps required.
+A `--force-rebuild` flag is also available for the command (and as a platform.sh runtime operation) to regenerate the database file from scratch if necessary.
+
+On PR and push, a separate GitHub action runs to check if the json file is valid, not empty, and that all the entries contain the mandatory keys.
+
+Users adding packages **must** include the following keys:
+
+- `packagistUrl`: mandatory, the information is collected by calling the Packagist API using this url
+- keywords: allows the user to define keywords to use as filters in the interface
+    - currently no limit set on either keyword length or number of keywords
+    - users can skip adding keywords by leaving the array empty
+- `usage` (`mezzio`, `mvc`): a mandatory single choice 'flag' that indicates a package's intended use
+    - any given value that does not match an option will cause the package to be skipped
+- `category`: mandatory, single choice field with the following values:
+    - `skeleton` -> usually 'project' type packages, providing an application base to build upon (e.g. mezzio-skeleton, which also inspired the name)
+    - `integration` -> packages that are based on, or extend others to bring further functionality (e.g. dot-sessionm extending laminas-session)
+    - `tool` -> standalone package that brings own functionality (e.g. laminas-filter)
+    - any given value that does not match an option will cause the package to be skipped
+
+- homepage: allows users to set a custom url for the homepage button in the interface
+    - if nothing is set by the user, the homepage is set from packagist if any is available, with the button skipped if none there as well
+
+#### Discussion
+
+Matthew suggests we need a usage option to indicate a "library" or "component" that can be used standalone, citing components like Tag and Feed and Paginator as examples. Aleksei suggests that categories should allow freeform words, similar to how keywords work in Composer; Julian says we can then make suggestions on categories, but not enforce them.
+
+Aleksei then asked why we need both categories and keywords. Julian explained it's to allow differentiating projects (which will be installed _on top of_ a framework), middleware (which can be installed in a pipeline), and libraries (which can be used standalone). Aleksei clarified further to application, tooling, and library.
+
+The final clarification:
+
+- Usage: application, tooling, library
+- Categories: freeform keywords
```

Comment:
I'm totally bikeshedding, but what about using the same terminology as composer's [type](https://getcomposer.org/doc/04-schema.md#type) and [keywords](https://getcomposer.org/doc/04-schema.md#keywords).

The vast majority of user would already be familiar with those terms and meanings. Even though the possible values for `type` would differ, I feel it would make it easier to understand what it is. It might also encourage to copy-paste composer keywords, which I think is a good thing.

```diff
The final clarification:

- - Usage: application, tooling, library
- - Categories: freeform keywords
+ - Type: application, tooling, library
+ - Keywords: freeform keywords
```